pkg/controller/job/job_controller.go

```diff
@@ -463,6 +463,16 @@ func (jm *JobController) patchJobResource(j *batch.Job, pods []*v1.Pod) error {
 	}
 
 	for _, pod := range pods {
+		// Skip if a resource update is in flight
+		if _, ok := pod.ObjectMeta.Annotations[schedulerapi.AnnotationResizeResourcesRequest]; ok {
```

Comment:
we need to understand how a pod update could trigger the job controller.

Reply:
I see the following call stack each time the pod is updated. Before I added the 'in-flight' check, it would repeatedly set resource-request annotation.

runtime/debug.Stack(0x56e8300, 0xc400000001, 0x37a89bf)
        /usr/local/go/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
        /usr/local/go/src/runtime/debug/stack.go:16 +0x22
k8s.io/kubernetes/pkg/controller/job.(*JobController).patchJobResource(0xc4212305a0, 0xc421ed0c00, 0xc420f447a0, 0x1, 0x1, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/job/job_controller.go:467 +0x353
k8s.io/kubernetes/pkg/controller/job.(*JobController).syncJob(0xc4212305a0, 0xc420a8ed00, 0xd, 0xc4229bbd00, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/job/job_controller.go:585 +0x14c5
k8s.io/kubernetes/pkg/controller/job.(*JobController).(k8s.io/kubernetes/pkg/controller/job.syncJob)-fm(0xc420a8ed00, 0xd, 0xc420a19d80, 0x2d926a0, 0xc421f44070)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/job/job_controller.go:137 +0x3e
k8s.io/kubernetes/pkg/controller/job.(*JobController).processNextWorkItem(0xc4212305a0, 0xc421fcd300)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/job/job_controller.go:393 +0xd1
k8s.io/kubernetes/pkg/controller/job.(*JobController).worker(0xc4212305a0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/job/job_controller.go:382 +0x2b
k8s.io/kubernetes/pkg/controller/job.(*JobController).(k8s.io/kubernetes/pkg/controller/job.worker)-fm()
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/job/job_controller.go:155 +0x2a
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4229a4160)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc4229a4160, 0x3b9aca00, 0x0, 0xc420472801, 0xc420472840)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc4229a4160, 0x3b9aca00, 0xc420472840)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by k8s.io/kubernetes/pkg/controller/job.(*JobController).Run
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/job/job_controller.go:155 +0x192
W1013 00:42:20.213734       1 job_controller.go:474] A resource update is in progress for pod sdjob-hbt6w. Skipping pod.

Reply:
yes however this doesn't explain why a pod update would wake up job controller